### ISSUE TITLE
Adapt polyfill to current state of decimal proposal?

### ISSUE THREAD

**Issue by jessealama** [Yaffle/BigDecimal#16]
Jesse here, one of the co-champions of the decimal proposal! As I am preparing to give an update to the TC39 plenary, I noticed your polyfill. This looks great! There was a time when the champions were considering an arbitrary-precision model for decimals, but after a lot of discussion we've settled on a fixed bit-width approach, IEEE 754 Decimal128 in particular. Would you be interested in updating your polyfill to reflect those changes? FWIW I have an [npm library](https://www.npmjs.com/package/decimal128) that implements Decimal128 (or, more precisely, the part of Decimal128 that's relevant to the proposal as it currently stands).

**Comment by Yaffle** [comment]
Hello Jesse! It's great to hear from you.
Some thoughts:
I see, that operator overloading will not be used, how will the interface look like?
My library now looks like:
 BigDecimal.BigDecimal(bigint)
 BigDecimal.BigDecimal(string)
 BigDecimal.BigDecimal(number) (only integers)
 BigDecimal.toBigInt(a) (not in the spec)
 BigDecimal.toNumber(a) (not in the spec, only integers)
 BigDecimal.unaryMinus(a)
 BigDecimal.add(a, b[, rounding])
 BigDecimal.subtract(a, b[, rounding])
 BigDecimal.multiply(a, b[, rounding])
 BigDecimal.divide(a, b, rounding)
 BigDecimal.lessThan(a, b)
 BigDecimal.greaterThan(a, b)
 BigDecimal.equal(a, b)
 BigDecimal.round(a, rounding)
 a.toString()
 a.toFixed(fractionDigits[, roundingMode = "half-up"])
 a.toPrecision(precision[, roundingMode = "half-up"])
 a.toExponential(fractionDigits[, roundingMode = "half-up"])
 BigDecimal.log(a, rounding)
 BigDecimal.exp(a, rounding)
 BigDecimal.sin(a, rounding)
 BigDecimal.cos(a, rounding)
 BigDecimal.atan(a, rounding)
 BigDecimal.sqrt(a, rounding)
 BigDecimal.abs(a)
 BigDecimal.sign(a)
 BigDecimal.max(a, b)
 BigDecimal.min(a, b)
And I cannot find the "interface" in the spec, but I vote for something much shorter, like `dec128.mul` or `d128.mul`, otherwise it may look too bulky.
What are the performance requirements? well, if it is 110 bits, then perhaps, internal usage of the BigInt to store the significand is not bad. Because Bigint has some overhead, and double-double arithmetic could be faster in case 106 bits are needed. Well... may be there are some other smart ways to implement it.

**Comment by Yaffle** [comment]
Ah, I see the spec now

**Comment by Yaffle** [comment]
@jessealama 
what should .coefficient/.exponent return when the value is +-0, +-Infinity, NaN ?

**Comment by Yaffle** [comment]
@jessealama , why do you use Decimal.cmp not Decimal.compare if you have long names for everything else

**Comment by Yaffle** [comment]
@jessealama , I have done some changes, but there are still a lot of work if to support multiplyAndAdd, reminder, and numbers without normalization

**Comment by Yaffle** [comment]
@jessealama , non-normalized decimals looks pretty natural, well. i have not tested all cases

**Comment by Yaffle** [comment]
@jessealama I have run the tests from https://github.com/jessealama/decimal128/tree/main/tests/dectest against my code and fixed few bugs, I see some tests fails when the result is rounded to infinity or not rounded, not sure if it a bug of test suite or my code's bug

**Comment by jessealama** [comment]
Awesome! Thanks for taking a look. Yes, supporting non-normalized values is a bit tricky. Is there anything you need some help with? I'd be happy to take a look at a PR, for instance.

**Comment by jessealama** [comment]
> @jessealama what should .coefficient/.exponent return when the value is +-0, +-Infinity, NaN ?

Those properties were intended to be private -- no need to worry about that!

**Comment by jessealama** [comment]
There's just `toString`, no more `toFixed` or `toExponential`.

**Comment by munrocket** [comment]
> I vote for something much shorter, like dec128.mul or d128.mul, otherwise it may look too bulky.

> why do you use Decimal.cmp not Decimal.compare if you have long names for everything else

Absolutely agree with it, filled the same ideas here https://github.com/tc39/proposal-decimal/issues/105
This is great that proposal become easy to polyfill!

**Comment by Yaffle** [comment]
> Awesome! Thanks for taking a look. Yes, supporting non-normalized values is a bit tricky. Is there anything you need some help with? I'd be happy to take a look at a PR, for instance.

not entirely true, support for non-normalized values ​​according to the ІЕЕЕ754 standard seems quite natural, I didn’t even have to change the code, the only place is exact division:
For the sum: a+b the number of digits to keep is min(digits(a), digits(b));
For multiplication - sum(digits(a), d
